core/formats/sparse_bitmap.cpp

```diff
@@ -352,48 +454,55 @@ void sparse_bitmap_iterator::read_block_header() {
   const uint32_t popcnt = 1 + static_cast<uint16_t>(in_->read_short());
   index_ = index_max_;
   index_max_ += popcnt;
+  if (prev_doc_written_) {
+    prev_ = in_->read_int();  // last doc in a previously filled block
+  }
+
+  if (track_prev_doc_) {
+    std::get<seek_prev>(attrs_).reset(
+        [](const void* ctx) noexcept {
+          return *reinterpret_cast<const doc_id_t*>(ctx);
+        },
+        &prev_);
+  }
+
   if (popcnt == sparse_bitmap_writer::kBlockSize) {
     ctx_.all.missing = block_ - index_;
     cont_begin_ = in_->file_pointer();
 
-    seek_func_ = [](sparse_bitmap_iterator* self, doc_id_t target) {
-      std::get<document>(self->attrs_).value = target;
-      std::get<value_index>(self->attrs_).value = target - self->ctx_.all.missing;
-      return true;
-    };
+    seek_func_ = track_prev_doc_ ? &container_iterator<BT_RANGE, true>::seek
+                                 : &container_iterator<BT_RANGE, false>::seek;
   } else if (popcnt <= kBitSetThreshold) {
-    constexpr BlockType type = BT_SPARSE;
-    const size_t block_size = 2*popcnt;
+    const size_t block_size = 2 * popcnt;
     cont_begin_ = in_->file_pointer() + block_size;
     ctx_.u8data = in_->read_buffer(block_size, BufferHint::NORMAL);
-
-    // FIXME check alignment
-    seek_func_ = ctx_.u8data
-      ? &container_iterator<type>::seek<AT_DIRECT>
-      : &container_iterator<type>::seek<AT_STREAM>;
+    ctx_.sparse.index = index_;
+
+    ResolveType(
+        [this]<AccessType A, bool TrackPrev>() noexcept {
+          seek_func_ =
+              &container_iterator<BT_SPARSE, TrackPrev>::template seek<A>;
+        },
+        ctx_.u8data, track_prev_doc_);
```

Comment:
make for sparse_bitmap_iterator function: set_seek_func
```suggestion
    set_seek_func<BT_DENSE>();
```

set_seek_func:
```C++
template <BlockType Type>
void sparse_bitmap_iterator::set_seek_func() {
  auto impl = [&]<bool TrackPrev>() {
    if (ctx_.u8data) {
      return container_iterator<Type, TrackPrev>::template seek<AT_DIRECT>;
    } else {
      return container_iterator<Type, TrackPrev>::template seek<AT_STREAM>;
    }
  };

  // FIXME(gnusi): check alignment
  if (track_prev_doc_) {
    seek_func_ = &impl.template operator()<true>();
  } else {
    seek_func_ = &impl.template operator()<false>();
  }
}
```

Reply:
Done